Makefile

```diff
@@ -1,6 +1,21 @@
-.PHONY: all rust
+.PHONY: all rust go go-deps install-protoc clean
 
-all: rust
+all: rust go
 
 rust:
 	cargo build
+
+go: install-protoc install-protobuf-go-plugins go-deps
+	./scripts/generate-go.sh
+
+go-deps:
+	go mod download
+
+install-protoc:
```

Comment:
I suggest not to install protoc for users, at least not include it in our makefile. Users should be able to use their prefer way to install protoc. If we place protoc installation here, we are very likely to carry the burden to maintain it, which is tedious and unnecessary.

Reply:
You are right. As @yuanbohan suggests, I use the container `namely/protoc-all:1.51_1` to compile the code.